conans/client/tools/files.py

```diff
@@ -103,10 +105,22 @@ def print_progress(_, __):
             pass
 
     with zipfile.ZipFile(filename, "r") as z:
-        if not pattern:
-            zip_info = z.infolist()
-        else:
-            zip_info = [zi for zi in z.infolist() if fnmatch(zi.filename, pattern)]
+        zip_info = z.infolist()
```

Comment:
I was using `namelist` for `ZipFile` (and `getnames` for the `TarFile` similarly).
then, `os.path.commonprefix` on the returned names list.
if it is dot (`.`), then there are more than one file/folder on the root.